src/main/java/org/jabref/gui/keyboard/EmacsKeyBindings.java

```diff
@@ -0,0 +1,103 @@
+package org.jabref.gui.keyboard;
+
+import java.util.Optional;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.input.KeyEvent;
+
+import org.jabref.Globals;
+import org.jabref.logic.util.strings.EmacsStringManipulator;
+import org.jabref.model.util.ResultingEmacsState;
+import org.jabref.preferences.JabRefPreferences;
+
+public class EmacsKeyBindings {
+
+    public static void executeEmacs(Scene scene, KeyEvent event) {
+        boolean EmacsFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS);
+        if (EmacsFlag && scene.focusOwnerProperty().get() instanceof TextInputControl) {
+            boolean CAFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CA);
+            boolean CFFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CF);
+            boolean CNFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CN);
+            boolean AUFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_AU);
```

Comment:
What is the reason for asking for Alt+U? Is that a special JabRef functionality?

The reason for the Rebind check boxes is for users with mixed Emacs and Windows shortcuts. And I never remember me typing Alt+U as shortcut (but Ctrl+A for seledct-all very often - also Ctrl+F for "Find")